src/libstd/dynamic_lib.rs

```diff
@@ -263,6 +266,16 @@ mod dl {
                  symbol: *const libc::c_char) -> *mut libc::c_void;
         fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
     }
+
+    #[cfg(target_os = "openbsd")]
```

Comment:
`#[link_name="dl]` is no no-op but not for all target_os:
 - for OpenBSD I am sure (tested)
 - for FreeBSD and DragonFly, it should (according to the commit cited previously)
 - for others (linux, android, macos, ios): I don't known

The problem is the prototype of functions is the same, but under Linux (and some others), it comes from libdl, whereas under some others (freebsd, dragonfly, openbsd) it comes from libc (but freebsd & dragonfly define a stub libdl).

If I remove the `#[link_name="dl"]` (and the second block with `#[cfg(target_os="openbsd")]`), the build should fail under Linux (and some others). Or did I miss something ?